healer.lic

```diff
@@ -0,0 +1,139 @@
+=begin
+  Documentation:
+=end
+no_pause_all
+custom_require.call %w[events common common-arcana common-healing common-items]
+
+class Healer
+  include DRC
+  include DRCH
+  include DRCI
+  include DRCA
+
+  def empty_hands
+    echo 'Emptying hands' if debug?
+    @left = checkleft
+    echo "@left: #{@left}" if debug?
+    @right = checkright
+    echo "@right: #{@right}" if debug?
+    fput('stow left') if checkleft
+    fput('stow right') if checkright
+  end
+
+  def stop_training
+    Script.running.find_all { |s| !s.paused? && !s.no_pause_all && s.name != 'healme' }.each(&:pause)
+  end
+
+  def start_training
+    return if running?('healme')
+
+    fput("get my #{@right}") if @right
+    fput("get my #{@left}") if @left
+    @left = nil
+    @right = nil
+    Script.running.find_all { |s| s.paused? && !s.no_pause_all && s.name != 'healme' }.each(&:unpause)
+  end
+
+  def ready_for_patient
+    # return false if Script.running?('healme')
+    return false if @last_heal_check && Time.now - @last_heal_check < 10
+    health = check_health
+    @last_heal_check = Time.now
+    return false if health['diseased']
+    return false if health['poisoned']
+    return false if checkhealth < 70
+    return false unless health['wounds'].empty?
+    true
+  end
+
+  def heal_self
+    return if Script.running?('healme')
+    start_script('healme')
+  end
+
+  def heal_person
+    name = @queue.first
+    echo "Healing #{name}" if debug?
+    @patient = name
+    return if @touched && @touched >= Time.now
+
+    unless @touched
+      @touched = Time.now + 12
+      pause 1
+      waitrt?
+      fput("touch #{name}")
+      put("transfer #{name} vit quick")
```

Comment:
fput, not put

Reply:
This is from the original script. My understanding is that put is when we dont care about a response from the game. As someone using the script for several hours, this was never an issue.

Reply:
👍

Reply:
Appreciate the review nonetheless :) I think the argument on `put` for vitality is, 'we'll get it if we get it, its time to move on to wounds'